File: libnetwork/src/main/java/com/skymajo/libnetwork/JsonConvert.java
package com.skymajo.libnetwork;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;

import java.lang.reflect.Type;

public class JsonConvert implements Convert {
    @Override
    public Object convert(String reponse, Type type) {
        JSONObject jsonObject = JSON.parseObject(reponse);
        JSONObject data = jsonObject.getJSONObject("data");
        if (data != null) {
            Object dataObj = data.get("data");
            return JSON.parseObject(dataObj.toString(),type);
        }

        return null;
    }

    @Override
    public Object convert(String reponse, Class clz) {
        JSONObject jsonObject = JSON.parseObject(reponse);
        JSONObject data = jsonObject.getJSONObject("data");
        if (data != null) {
            Object dataObj = data.get("data");
            return JSON.parseObject(dataObj.toString(),clz);
        }
        return null;
    }
}

File: libnetcache/src/main/java/com/skymajo/libnetcache/JsonConvert.kt
package com.skymajo.libnetcache

import com.google.gson.Gson
import com.google.gson.JsonObject
import java.lang.reflect.Type

class JsonConvert<T>:Convert<T> {
    override fun convert(response: String, type: Type): T? {
        var jsonObject= Gson().fromJson(response,JsonObject::class.java)
        val data = jsonObject.getAsJsonObject("data")
        if(data!=null){
            return Gson().fromJson(data.get("data").toString(),type)
        }
        return null
    }

    override fun convert(response: String, claz: Class<*>): T? {
        var jsonObject=Gson().fromJson(response,JsonObject::class.java)
        var data=jsonObject.getAsJsonObject("data")
        data?.run {
            return Gson().fromJson(data["data"].toString(),claz) as T
        }
        return null
    }
}